### ISSUE TITLE
solc combined output not handled properly

### ISSUE THREAD

**Issue by chevdor** [ethcore/parity#3511]
Really cool what you implemented in #3196. I may do it wrong or I hit a little glitch.

I tested with a junk contract:
```
contract junk {
}
```

`solc --combined-json abi,bin junk.sol` produces the following output.

`{"contracts":{"junk":{"abi":"[]\n","bin":"6060604052600c8060106000396000f360606040526008565b600256"}},"version":"0.4.4+commit.4633f3de.Darwin.appleclang"}`

I pasted that in the abi/solc field and I saw the following good things:
- a listbox shows up to pick the contract
- the bytecode is extracted and filled up in the code field

What looks less good is that the abi/solc field still contains the output of solc which is no valid ABI.
Moreover, since solc provides a contract name, we could also populate the name field.

Since the name field is one of the first in the list, I would suggest to only auto-fill it from the solc output if it was empty so we don´t replace what the user may have just typed in.

I attach a screenshot, not that it is useful, just because Parity looks beautiful :)
![screen shot 2016-11-18 at 15 13 49](https://cloud.githubusercontent.com/assets/738724/20432913/b645a09a-ada1-11e6-9362-7d2abd3a763a.png)

In other words, the goal here should be to paste the solc output and be able to immediately click NEXT button.

**Comment by jacogr** [comment]
@ngotchac Some good feedback that I missed.

**Comment by chevdor** [comment]
This has been improved but I am not sure it is totally fixed yet:
![screen shot 2016-11-19 at 19 32 33](https://cloud.githubusercontent.com/assets/738724/20457594/fe81e5ec-ae8e-11e6-8d99-87ee31bc5fd1.png)

Look at the ABI, this is not just the ABI. It should be just [] in my mini sample.
Or is it then handled properly based on that string? If so, it would be fine.

**Comment by ngotchac** [comment]
Yes, the content of the input isn't changed, but when deployed it's the right ABI that is taken into account: here `[]`. You can verify this by deploying the contract, and click on the `Details` button on the top right of the contract view. The right ABI should be displayed :)
